internal/kafka/internal/services/quota/internal/utils/cloudaccount_billing_model_resolver.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	v1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+)
+
+/********************************************************************************************************************
+ * This resolver is used when the user specifies a `Cloud Account ID` in the request.
+ * The AMS Billing Model is then detected by checking what is the marketplace supported by that account.
+ ********************************************************************************************************************/
+
+var _ BillingModelResolver = &cloudAccountBillingModelResolver{}
+
+type cloudAccountBillingModelResolver struct {
+	quotaConfigProvider AMSQuotaConfigProvider
+	kafkaConfig         *config.KafkaConfig
+}
+
+func (r *cloudAccountBillingModelResolver) SupportRequest(kafka *dbapi.KafkaRequest) bool {
+	return kafka.BillingCloudAccountId != ""
+}
+
+func (r *cloudAccountBillingModelResolver) Resolve(orgId string, kafka *dbapi.KafkaRequest) (BillingModelDetails, error) {
+	kafkaBillingModel, err := r.kafkaConfig.GetBillingModelByID(kafka.InstanceType, "marketplace")
+	if err != nil {
+		return BillingModelDetails{}, err
+	}
+
+	return r.resolve(orgId, kafka, kafkaBillingModel)
+}
+
+// resolve - tries to resolve the AMS Billing Model by performing the following checks:
+// 1) Tries to retrieve the cloud account. A failure here will result in an error.
+// 2) Checks if the user has sufficient marketplace quota. If not, an error is returned.
+// 3) Tries to get the marketplace associated with the cloud provider.
```

Comment:
Shouldn't 3) be reworded adding "if not provided in the request"  at the end?

Reply:
Nope. This resolver runs when the `BillingCloudAccountId` is empty (see `SupportRequest`)

Reply:
The check in `SupportRequest` is when BillingCloudAccountId is not empty :thinking: :

```
	return kafka.BillingCloudAccountId != ""
```

Reply:
Sorry, I got confused. You are right. Moreover, the check on BillingCloudAccountId is not related to what you were asking (the marketplace).
I will update the comment.